components/Careers/AllJobsPage.vue

```diff
@@ -0,0 +1,427 @@
+                        "companyname": "Oracle Corporation",
+                        "location": "Japan",
+                        "joblink": "#6",
+                        "companyicon": ""
+                    },
+                    {
+                        "jobtitle": "Database Administrator",
+                        "daysafterposted": 2,
+                        "jobtype": "Contract",
+                        "salarystart": 1000,
+                        "salaryend": 1100,
+                        "salaryperiod": "wk",
+                        "companyname": "IBM Corporation",
+                        "location": "United States",
+                        "joblink": "#7",
+                        "companyicon": ""
+                    }
+                ],
+                currentPage: 1,
+                perPage: 6,
+            }
+        },
+        computed: {
+            sortedLocations() {
+                // Extract all locations from jobs data
+                const locations = this.jobsdata.map((job) => job.location);
+    
+                // Remove duplicates and sort the locations
+                return [...new Set(locations)].sort();
+            },
+            sortedTypes() {
+                // Extract all locations from jobs data
+                const jobtypes = this.jobsdata.map((job) => job.jobtype);
+    
+                // Remove duplicates and sort the locations
+                return [...new Set(jobtypes)].sort();
+            },
+            filteredJobs() {
+                let filteredJobs = this.jobsdata;
+    
+                if (this.filterdata.keyword != "") {
+                    filteredJobs = filteredJobs.filter((job) => {
+                        return job.jobtitle.toLowerCase().includes(this.filterdata.keyword.toLowerCase());
+                    });
+                }
+    
+                if (this.filterdata.location != 'none') {
+                    filteredJobs = filteredJobs.filter((job) => {
+                        return job.location === this.filterdata.location;
+                    });
+                }
+    
+                if (this.filterdata.type != 'none') {
+                    filteredJobs = filteredJobs.filter((job) => {
+                        return job.jobtype === this.filterdata.type;
+                    });
+                }
+    
+                return filteredJobs;
+            },
+            totalPages() {
+                const total = Math.ceil(this.filteredJobs.length / this.perPage);
+                console.log(this.filteredJobs.length, this.perPage)
+                return total;
+            },
+    
+        },
+        methods: {
+            getPaginatedData() {
+                const start = (this.currentPage - 1) * this.perPage;
+                const end = start + this.perPage;
+                const paginated = this.filteredJobs.slice(start, end);
+                return paginated;
+            },
+            subitHandler(){
```

Comment:
fix the typo, it should be submitHandler
also its not submit but search button clicked 
please give name according to its functionality